### PR TITLE
feat(controlplane): disallow non-alphanumeric names

### PR DESCRIPTION
**Pull request by jordivilaseca**
## Describe your changes
Force names to be alphanumeric with `.` , `/`,  `(`,  `)`, `-`, `_` characters. The validation is performed server side in the controlplane

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have made the appropriate changes to the documentation
- [x] New and existing unit tests pass locally with my changes
- [x] If it is a major user facing functionality, I have added behavioural tests
- [x] If it is a critical part of the code or of significant complexity, I have added thorough testing

### CODE REVIEW

File: cmd/neblictl/internal/controlplane/executors.go [hunk 1]
```diff
@@ -270,7 +270,7 @@ func (e *Executors) StreamsCreate(ctx context.Context, parameters interpoler.Par
 		// Propagate new configuration
 		err := e.controlPlaneClient.setSamplerConfig(ctx, resourceAndSamplerEntry.sampler, resourceAndSamplerEntry.resource, update)
 		if err != nil {
-			writer.WriteStringf("%s.%s: Could not create the stream because %v\n", resourceAndSamplerEntry.resource, resourceAndSamplerEntry.sampler, err)
+			writer.WriteStringf("%s.%s: ERROR %v\n", resourceAndSamplerEntry.resource, resourceAndSamplerEntry.sampler, err)
```
Comment (GerardGarcia) [inline cmd/neblictl/internal/controlplane/executors.go:273]: This significantly changes the error message format, we do not have a "standard" defined but putting the error level in upper case only in a few cases makes these seem like they are specially bad errors. All changes to the error messages should be consistent and I probably deserve their own PR

File: controlplane/client/client.go [hunk 2]
```diff
@@ -138,7 +138,7 @@ func (c *Client) ConfigureSampler(ctx context.Context, samplerResource, samplerN
 
 	status := samplerConfRes.SamplerConfRes.GetStatus()
 	if status.GetType() != protos.Status_OK {
-		return fmt.Errorf("error configuring rule rules: %s", status.GetErrorMessage())
+		return fmt.Errorf("%s", status.GetErrorMessage())
```
Comment (GerardGarcia) [inline controlplane/client/client.go:141]: similarly, in the previous method we have the prefix `error getting samplers list`, why in this case we should not have it, and if so, if there isn't any good reason it should be consistent with the other errors.

File: controlplane/server/internal/protocol/client/sampler.go [hunk 3]
```diff
@@ -8,6 +8,10 @@ import (
 	"github.com/neblic/platform/controlplane/server/internal/defs"
 )
 
+func validateUID() {
```
Comment (GerardGarcia) [inline controlplane/server/internal/protocol/client/sampler.go:11]: empty function?

File: controlplane/control/sampler.go [hunk 4]
```diff
@@ -1,11 +1,16 @@
...
 import (
+	"errors"
 	"fmt"
+	"regexp"
 
 	"github.com/neblic/platform/controlplane/protos"
 )
 
+var uidValidationRegex = regexp.MustCompile(`^[\.\/\ ()\w-]*$`)
+var uidValidationErrTemplate = "invalid %s uid %s, expected alphanumerical with spaces and ./() characters"
```
Comment (GerardGarcia) [inline controlplane/control/sampler.go:12]: * Should we allow spaces? I think it is dangerous to have an uid with spaces
* The comment doesn't say `-` and `_` are allowed.

File: cmd/neblictl/internal/controlplane/executors.go (hunk 1 above)
Comment (jordivilaseca) [reply cmd/neblictl/internal/controlplane/executors.go:273]: In case of error, neblictl showed something like `kafka-sampler.order-confirmation: Could not create the stream because error configuring the rule rules: invalid %s uid %s, expected alphanumerical with spaces and ./() character`. Half of the error is boilerplate that does not provide any useful information. In fact, it's misleading, it talks about rules, when it does not have anything to do with it. As different kind of errors can happen, and in multiple contexts (digests, streams and events), I thought it would be simpler to show the inner error without adding boilerplate.

File: controlplane/client/client.go (hunk 2 above)
Comment (jordivilaseca) [reply controlplane/client/client.go:141]: `error configuring rule rules:` prefix was always returned. When it was firstly introduced, the oly thing it could be created where streams, and the only possible error a rule error. But now we have multiple types of errors and primitives to be configured and the error does not make sense. I could change it to another one, but it will need to be really general, I think it's better that the error function contains all the error information (it will know the context and why it happened)

File: controlplane/server/internal/protocol/client/sampler.go (hunk 3 above)
Comment (jordivilaseca) [reply controlplane/server/internal/protocol/client/sampler.go:11]: Good catch, will remove that

File: controlplane/control/sampler.go (hunk 4 above)
Comment (jordivilaseca) [reply controlplane/control/sampler.go:12]: - True, will remove the spaces
- I just updated the description to add those characters (and remove the spaces)

**Review by GerardGarcia — APPROVED** [review]
